src/components/AnalysisPage/ThicknessAnalysisBox/ThickAnalysisBox.jsx

```diff
@@ -61,6 +62,19 @@ const ThickAnalysisBox = ({
     );
   };
 
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      const container = listOfThicknessDataContainerRef.current;
+      if (container && container.scrollHeight > container.clientHeight) {
+        container.scrollBy({ top: 100, behavior: "smooth" });
+        setTimeout(() => {
+          container.scrollBy({ top: -100, behavior: "smooth" });
+        }, 800);
+      }
+    }, 1000);
+    return () => clearTimeout(timer);
```

Comment:
What are these 2 setTimeouts doing? Is it just adding an animation type effect?

Reply:
Yeah they are for the scrolling animation of the threshold history container when there is more than 1 row. 

it waits a second to start the animation, then the scroll bar moves to the bottom while the setTimeout is waiting 800ms to run and bring the scroll bar to its original spot

Reply:
Ok sweet thats probably cool. Is it possible to get it to happen right now? It looks like this gets triggered when a new part is loaded in but it currently loads in the part with no units selected so I dont think this will ever happen. Although I know that I made a comment below to change it so that parts are loaded in with units so this might work again after you change that. But also should this useEffect be triggered when the units are changed now not when the partId is changed since the thresholdHistory depends on the units selected for a part?

Reply:
I just tested it out and you're correct, It'll never happen. 

But after I make the change for the parts to be loaded with units, I think maybe we would want this effect to run when units change and also when a new part is loaded.. 

Because i'm thinking that maybe if a new part has the same units as the previous part it might not run because nothing actually changed? 

or maybe it will run because units gets set to null for a brief second. 

The answer is coming soon

Reply:
Yeah this works and it works better.

Before if you switched between mm and inches and they both had more than 1 row. you wouldn't get the effect on that switch. Now you get the effect on the load and the switch even if the previous part had the same units